prow/jenkins/controller.go

```diff
@@ -413,12 +437,16 @@ func (c *Controller) syncTriggeredJob(pj prowapi.ProwJob, reports chan<- prowapi
 		if err != nil {
 			return fmt.Errorf("error getting build ID: %v", err)
 		}
+		linkURL, err := c.getStatusErrorLink()
+		if err != nil {
+			return fmt.Errorf("error getting status link: %v", err)
+		}
 		// Start the Jenkins job.
 		if err := c.jc.Build(&pj, buildID); err != nil {
 			c.log.WithError(err).WithFields(pjutil.ProwJobFields(&pj)).Warn("Cannot start Jenkins build")
 			pj.SetComplete()
 			pj.Status.State = prowapi.ErrorState
-			pj.Status.URL = testInfra
+			pj.Status.URL = linkURL.String()
```

Comment:
If we only ever need the string we don't need to store the URL but should validate that the value in config is a URL when loading